server/models/customer.js

```diff
@@ -33,3 +32,120 @@ const orderSchema = new mongoose.Schema({
       type: mongoose.Schema.Types.Decimal128
     }
 });
+
+const basketSchema = mongoose.Schema({
+        customerId: {
+            type: mongoose.Schema.Types.ObjectId,
+            ref: 'Customer',
+            required: true,
+            unique: true 
```

Comment:
please remove the customerId from the the basket as the basket can be only accessible through the customerId of the customer schema, so adding the customerId here would be redundunt.